src/files/modals/publish-modal/PublishModal.js

```diff
@@ -0,0 +1,157 @@
+import React, { useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
+import { withTranslation } from 'react-i18next'
+import { CopyToClipboard } from 'react-copy-to-clipboard'
+import Button from '../../../components/button/Button'
+import { Modal, ModalActions, ModalBody } from '../../../components/modal/Modal'
+import Icon from '../../../icons/StrokeSpeaker'
+import { connect } from 'redux-bundler-react'
+import Radio from '../../../components/radio/Radio'
+import ProgressBar from '../../../components/progress-bar/ProgressBar'
+import './PublishModal.css'
+
+const PublishModal = ({ t, tReady, onLeave, onSubmit, file, ipnsKeys, publicGateway, className, doFetchIpnsKeys, doUpdateExpectedPublishTime, expectedPublishTime, ...props }) => {
+  const [disabled, setDisabled] = useState(true)
+  const [error, setError] = useState(null)
+  const [selectedKey, setSelectedKey] = useState({ name: '', id: '' })
+  const [link, setLink] = useState('')
+  const [progress, setProgress] = useState(0)
+  const [start, setStart] = useState(null)
+
+  useEffect(() => {
+    setDisabled(selectedKey.name === '' || start !== null)
+  }, [selectedKey, start])
+
+  useEffect(() => {
+    doFetchIpnsKeys()
+  }, [doFetchIpnsKeys])
+
+  useEffect(() => {
+    if (!start) return
+
+    const interval = setInterval(() => {
```

Comment:
:thought_balloon:  nit: I wonder if we could remove `setInterval` here and use [css animations](https://stackoverflow.com/question/15945288/animating-progress-bars-with-css) instead somehow?  (would look less choppy, we could also make it more engaging by making it non-linear). 

I'm thinking setting `progress` to 100% and using  `animation: progressBar ${expectedPublishTime}s ease-in-out`  for customizing the progress animation instead.

Just an idea, feel free to ignore.